src/comm/aca_message_pulsar_consumer.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_message_pulsar_consumer.h"
+#include "goalstate.pb.h"
+#include "aca_comm_mgr.h"
+#include "aca_log.h"
+
+using aca_comm_manager::Aca_Comm_Manager;
+using pulsar::Client;
+using pulsar::ClientConfiguration;
+using pulsar::ConsumerConfiguration;
+using pulsar::Consumer;
+using pulsar::Message;
+using pulsar::Result;
+
+
+namespace aca_message_pulsar
+{
+ACA_Message_Pulsar_Consumer::ACA_Message_Pulsar_Consumer(string brokers, string subscription_name)
+{
+  setBrokers(brokers);
+  setSubscriptionName(subscription_name);
+
+  // TO DO: Construct the configuration
+
+  ACA_LOG_DEBUG("Broker list: %s\n", this->brokers_list.c_str());
+  ACA_LOG_DEBUG("Consumer subscription name: %s\n", this->subscription_name.c_str());
+
+  // Create the consumer and client
+  this->ptr_consumer = new Consumer();
+  this->ptr_client= new Client(brokers,this->client_config);
+}
+
+ACA_Message_Pulsar_Consumer::~ACA_Message_Pulsar_Consumer()
+{
+  //delete ptr_consumer;
```

Comment:
looks like we want to clean up ptr_consumer also right?

Reply:
We do need to clean up ptr_consumer but there would be a warning when I do so cause the consumer class has no explicit destructor.

Reply:
I see, looks like we will use c++ mechanism to clean up when it go out of scope (or when ACA exits). Please add comment for the intent and explain why it is commented out.